JH5/task10/src/homework/HtmlCreator.java

```diff
@@ -0,0 +1,99 @@
+package homework;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Class HtnlCreator creates html code and returns string of html code
+ */
+public class HtmlCreator {
+
+    /**
+     * Colors of html-table
+     */
+    private static final String HEAD_COLOR = "#999999";
+    private static final String COL1_COLOR = "#efefef";
+    private static final String COL2_COLOR = "#D9D9D9";
+    private static final String COL_MAX_COLOR = "FF0000";
+    private static final String BORDER_COLOR = "#ffffff";
+
+    /**
+     * String HEADER contains header of html code
+     */
+    private static final String HEADER = "<html>\n " +
+            "<head>\n " +
+            "<title>Table</title>\n " +
+            "<meta charset = \"utf-8\">\n " +
+            "</head>\n " +
+            "<body>\n" +
+            "<table border=1 bordercolor =\"" + BORDER_COLOR + "\"CELLSPACING = 0" +
+            "cellpadding=\"1\"  width=\"630\" height=\"230\" align = \"center\">\n " +
+            "<tr bgcolor=\"" + HEAD_COLOR + "\"align = center valign = center>\n " +
+            "<th width=\"630\">Server</th>\n " +
+            "<th width=\"630\">Response, ms</th>\n";
+
+    /**
+     * String FOOTER contains finish part of html code
+     */
+    private static final String FOOTER = "</table>\n </body>\n </html>\n";
+
+    /**
+     * Body of the html code
+     */
+    private String HTML = "";
+
+    /**
+     * Method getHtml returns full html code of html-table
+     */
+    public String getHtml() {
+        return HTML;
+    }
+
+    /**
+     * Constructor HtmlCreator gets collection of ip addresses and creates
+     * a full html code of html table.
+     *
+     * @param addresses collection of ip addresses
+     */
+    public HtmlCreator (HashMap<String, Integer> addresses) {
+        int max = 0;
+        // Searching for the maximal time
+        for (Map.Entry<String, Integer> ip : addresses.entrySet()) {
+            if (ip.getValue() > max) {
+                max = ip.getValue();
+            }
+        }
+        // Create another line of table
+        setLine(addresses, max);
+        // Create full html code
+        HTML = HEADER + HTML + FOOTER;
+    }
+
+    /**
+     * Method setLine gets collection of ip addresses and maximal time,
+     * depending on maximal time set color of line, creates another line of table.
+     *
+     * @param addresses collection of ip addresses
+     * @param max maximal time of request
+     */
+    private void setLine(HashMap<String, Integer> addresses, int max) {
```

Comment:
You already iterated through addresses in constructor - why are you iterating again in setLine? I suppose setLine must know only about one server (table row). Please refactor the code to have only one cycle. And  on my mind this cycle must be placed out of HtmlCreator